pkg/docker/config/config.go

```diff
@@ -587,6 +614,16 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 		return getAuthFromCredHelper(ch, registry)
 	}
 
+	// Support for different paths in auth.
+	if !legacyFormat && ref != nil {
+		for _, key := range authKeysForRef(ref) {
+			if val, exists := auths.AuthConfigs[key]; exists {
+				return decodeDockerAuth(val)
+			}
+		}
+	}
+
+	// Fallback to the registry based lookup.
```

Comment:
I think this could be moved into an `} else {` branch above.

Reply:
Yep, fixed as suggested :+1: